crates/dodgy_3d/src/lib.rs

```diff
@@ -0,0 +1,477 @@
+#![doc = include_str!("../README.md")]
+// The contents of this file were primarily ported from Agent.cc from RVO2-3D
+// with significant alterations. As per the Apache-2.0 license, the original
+// copyright notice has been included, excluding those notices that do not
+// pertain to the derivate work:
+//
+// Agent.cc
+// RVO2 Library
+//
+// SPDX-FileCopyrightText: 2008 University of North Carolina at Chapel Hill
+//
+// The authors may be contacted via:
+//
+// Jur van den Berg, Stephen J. Guy, Jamie Snape, Ming C. Lin, Dinesh Manocha
+// Dept. of Computer Science
+// 201 S. Columbia St.
+// Frederick P. Brooks, Jr. Computer Science Bldg.
+// Chapel Hill, N.C. 27599-3175
+// United States of America
+//
+// <https://gamma.cs.unc.edu/RVO2/>
+mod linear_programming;
+mod simulator;
+
+use crate::linear_programming::{solve_linear_program, Plane};
+
+pub use glam::Vec3;
+pub use simulator::{AgentParameters, Simulator, SimulatorMargin};
+
+// A single agent in the simulation.
+#[derive(Clone, PartialEq, Debug)]
+pub struct Agent {
+  // The position of the agent.
+  pub position: Vec3,
+  // The current velocity of the agent.
+  pub velocity: Vec3,
+
+  // The radius of the agent. Agents will use this to avoid bumping into each
+  // other.
+  pub radius: f32,
+  // The maximum velocity the agent is allowed to move at.
+  pub max_velocity: f32,
```

Comment:
max_velocity can be move to AgentParameters in the Simulator, and moved as an argument to compute_avoidance_velocity here, since it's only needed on "self"

Reply:
I'm just thinking it is a good idea to minimize the size of this, in case one wants to keep them around

Reply:
Done. This was a great suggestion! I've added a commit for it.